src/main/java/backend/Logic.java

```diff
@@ -231,9 +232,24 @@ public void removeUnusedModels(Set<String> reposInUse) {
                 .forEach(models::removeRepoModelById);
     }
 
-    public ImmutablePair<Integer, Long> updateRemainingRate(ImmutablePair<Integer, Long> rateLimits) {
-        uiManager.updateRateLimits(rateLimits);
-        return rateLimits;
+    /**
+     * Updates the GitHub API quota info in the GUI.
+     * @param apiQuotaInfo The GitHub API quota information.
+     * @return the apiQuotaInfo instance
+     */
+    public ApiQuotaInfo updateApiQuotaInfoInGui(ApiQuotaInfo apiQuotaInfo) {
+        uiManager.updateApiQuotaInfo(apiQuotaInfo);
+        return apiQuotaInfo;
```

Comment:
Oh ok. thenAccept works. However thenAccept can only be used once for each thenCompose. I need to run two methods that are using the thenCompose values.

if i use thenAccept, i need to do something like this. Running thenCompose twice.
```
                .thenCompose(n -> getRateLimitResetTime())
                .thenAccept(this::updateSyncRefreshRate)
                .thenCompose(n -> getRateLimitResetTime())
                .thenAccept(this::updateApiQuotaInfoInGui)

```

This two method don't get called together all the time, so i can't move them into a single method.

So for efficiency i guess thenApply should be ok?